### DISCUSSION TITLE
`This workflow does not exist.` is confusing and has all sorts of ramifications

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#16551] [Actions]
<details><summary>Confusing things happen when a workflow ceases to be present on the default branch</summary>

## Fork a repository / Enable Actions / Trigger Actions
I have a repository that's a fork of another repository, and I've enabled actions in my fork:
https://github.com/jsoref/kubernetes-ingress/actions
<img width="619" alt="image" src="https://user-images.githubusercontent.com/2119212/168347538-86427e6d-2a52-4804-bc71-7c5db8be6940.png">

As you can see, it has run a lot of actions.

### CI Action Run
The second row in that list is this:
https://github.com/jsoref/kubernetes-ingress/actions/runs/2317476464
<img width="1418" alt="image" src="https://user-images.githubusercontent.com/2119212/168347691-ee5c7d70-bfa1-4b36-ade7-89a83bd35a8e.png">
Which has this workflow file:
https://github.com/jsoref/kubernetes-ingress/actions/runs/2317476464/workflow
<img width="481" alt="image" src="https://user-images.githubusercontent.com/2119212/168347783-6cf7797f-071b-4383-9ca4-542fef82ad13.png">
specifically this workflow:
https://github.com/jsoref/kubernetes-ingress/blob/bf0e5baa66568d3adde1c4ccd989e3b93a06a421/.github/workflows/ci.yml

### CI Action overview

Now, classically, you could load this url, which roughly corresponds to the preceding workflow file...
https://github.com/jsoref/kubernetes-ingress/actions/workflows/ci.yml

And right now, I get this:
<img width="575" alt="image" src="https://user-images.githubusercontent.com/2119212/168347910-7b7da4b6-3738-4d74-9618-afd6bb25d1aa.png">

This is at least partially a lie. And it's definitely misleading, and it's absolutely unhelpful.

### How did we get here

The fun thing is that this file used to exist on the default branch, until I added a new branch with no workflows and changed the default branch to point to it: 
https://github.com/jsoref/kubernetes-ingress/commits/placeholder
https://github.com/jsoref/kubernetes-ingress/commit/d1d67f5d9faabba1ec01c0986c1ccd623e87bd20
</details>

## Problems
1. It should be possible to see all workflows that were run by a pathed workflow even after it ceases to exist on the default branch.
2. It should be possible to know if a workflow is disabled even if it is no longer present on the default branch (from memory the answer is that disabling a workflow and then removing it from the default branch clears the disable flag, but this is not remotely discoverable and I believe not documented. It resulted in a huge security problem for people consuming my workflow, and I really want to speak to an engineer/PM about this problem, but that's mostly out of scope).
3. It's at least partially inaccurate and definitely misleading, the workflow exists in that it has created runs that a user would legitimately want to see, and in fact could see w/o filtering.
   * You could fix the misleading nature of the message by changing it to:
     > This workflow is not present on the default branch.

**Comment by zhangjing-GitHub-Code** [comment]
@orgs @github @community Hey, anyone to solve this OBVIOUSLY IMPORTANT AND **ING PROBLEM?
My repo main is updated 3yrs ago, but it DO HAVE .github/worflow/xxx.yml, and if I click on 'View Runs', The incoming page says: This workflow does not exist.

**Reply by jsoref** [reply]
I think you're tripping on a _different_ problem, namely that GitHub only stores runs in public repositories for 90 days. [Configuring the retention period for GitHub Actions artifacts and logs in your organization](https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization)

The view only really counts workflow runs that it can see (up to 90 days for public repositories and up to 400 days for private repositories depending on private repository configuration).

It shouldn't say:
> "This workflow does not exist."

A correct message would be:
> "This workflow has not run in the last x days (see [Configuring the retention period for GitHub Actions artifacts and logs in your organization](https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization))."

And, of course, since the workflow exists, it should provide a way to view the workflow... It shouldn't require a recent run to get access to the workflow file.
